app/graphql/types/wiki_page_edit_type.rb

```diff
@@ -7,10 +9,41 @@ class WikiPageEditType < Types::BaseObject
     field :user, UserType, null: true
     field :diff, String, null: true
     field :diff_html, String, null: true
+    field :change_set, WikiPageEditsType, null: false
     field :comment, String, null: true
 
     def user
       Loaders::Association.for(WikiPageEdit, :user).load(object)
     end
+
+    def change_set
+      parsed = Nokogiri::HTML.parse(object.diff_html)
+      body_changed = false
+      fm_changed = false
+      in_front_matter = false
+      tags = parsed.xpath("//li").map do |tag|
+        front_matter = false
+        body = false
+
+        in_front_matter = !in_front_matter if tag.text == "---"
```

Comment:
Since we can rely on front matter existing at the top of the document, and only existing there, we implement a naive parsing mechanism to serially evaluate whether we're in the context of front matter to determine the source of a given change.

Reply:
I don't have a gut feeling for how fast Nokogiri is.  Down the road if a study has tens or hundreds of diffs do think this could turn into a performance problem?

Reply:
For tens of diffs this will probably be fine. We have lots of runway for where this will start to become a performance issue. At that point in time, we can limit the number of wiki edits we display by default.